docs/developer/adr/0001-remove-isinstance-checks-when-setting-parameters.md

```diff
@@ -23,13 +23,38 @@ Therefore, we should remove this mechanism and replace it with a more general va
 The more general validation mechanism can be considered out of scope for Sciline, and should be implemented in the user code or using other common libraries such as `pydantic`.
 
 Finally, we can think of this mechanism as a form of runtime type checking.
-This is not the intended scope of Sciline.
+We should ask ourselves if this is the intended scope of Sciline.
+If it is, shouldn't we also check that each provider actually returns the correct type?
```

Comment:
Can you reformulate to turn it into a statement that this is not the scope of sciline?

Reply:
I guess I should put that statement into the "Decision" section?

Reply:
Yes